From f25896558d8c9dd5b1c001b98f499339bf896c37 Mon Sep 17 00:00:00 2001
From: Rodrigo Agerri <ragerri@apache.org>
Date: Wed, 8 Oct 2014 16:22:31 +0000
Subject: [PATCH] OPENNLP-717 bug fixed by giving access to feature generator
 in TokenNameFinderFactory for NameFinder model creation in the NameFinderME
 train method

git-svn-id: https://svn.apache.org/repos/asf/opennlp/trunk@1630163 13f79535-47bb-0310-9956-ffa450edef68
---
 .../java/opennlp/tools/namefind/NameFinderME.java     |  4 ++--
 .../tools/namefind/TokenNameFinderFactory.java        | 11 ++++-------
 2 files changed, 6 insertions(+), 9 deletions(-)

diff --git a/opennlp-tools/src/main/java/opennlp/tools/namefind/NameFinderME.java b/opennlp-tools/src/main/java/opennlp/tools/namefind/NameFinderME.java
index a2131f745..91f487a6b 100644
--- a/opennlp-tools/src/main/java/opennlp/tools/namefind/NameFinderME.java
+++ b/opennlp-tools/src/main/java/opennlp/tools/namefind/NameFinderME.java
@@ -355,10 +355,10 @@ else if (TrainerType.EVENT_MODEL_SEQUENCE_TRAINER.equals(trainerType)) {
     }
 
     if (seqModel != null) {
-      return new TokenNameFinderModel(languageCode, seqModel, null,
+      return new TokenNameFinderModel(languageCode, seqModel, factory.getFeatureGenerator(),
               factory.getResources(), manifestInfoEntries, factory.getSequenceCodec());
     } else {
-      return new TokenNameFinderModel(languageCode, nameFinderModel, beamSize, null,
+      return new TokenNameFinderModel(languageCode, nameFinderModel, beamSize, factory.getFeatureGenerator(),
               factory.getResources(), manifestInfoEntries, factory.getSequenceCodec());
     }
   }
diff --git a/opennlp-tools/src/main/java/opennlp/tools/namefind/TokenNameFinderFactory.java b/opennlp-tools/src/main/java/opennlp/tools/namefind/TokenNameFinderFactory.java
index cd4792bee..8d4f41d47 100644
--- a/opennlp-tools/src/main/java/opennlp/tools/namefind/TokenNameFinderFactory.java
+++ b/opennlp-tools/src/main/java/opennlp/tools/namefind/TokenNameFinderFactory.java
@@ -21,20 +21,13 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Map;
-import java.util.Properties;
 
-import opennlp.tools.chunker.ChunkerContextGenerator;
-import opennlp.tools.dictionary.Dictionary;
 import opennlp.tools.namefind.TokenNameFinderModel.FeatureGeneratorCreationError;
-import opennlp.tools.postag.POSTaggerFactory;
-import opennlp.tools.postag.TagDictionary;
 import opennlp.tools.util.BaseToolFactory;
 import opennlp.tools.util.InvalidFormatException;
 import opennlp.tools.util.SequenceCodec;
-import opennlp.tools.util.SequenceValidator;
 import opennlp.tools.util.ext.ExtensionLoader;
 import opennlp.tools.util.featuregen.AdaptiveFeatureGenerator;
-import opennlp.tools.util.featuregen.AdditionalContextFeatureGenerator;
 import opennlp.tools.util.featuregen.AggregatedFeatureGenerator;
 import opennlp.tools.util.featuregen.FeatureGeneratorResourceProvider;
 import opennlp.tools.util.featuregen.GeneratorFactory;
@@ -75,6 +68,10 @@ protected SequenceCodec<String> getSequenceCodec() {
   protected Map<String, Object> getResources() {
     return resources;
   }
+  
+  protected byte[] getFeatureGenerator() {
+    return featureGeneratorBytes;
+  }
 
   public static TokenNameFinderFactory create(String subclassName, byte[] featureGeneratorBytes, final Map<String, Object> resources,
       SequenceCodec<String> seqCodec)
